scenarios/cpp/windows/captioning/caption.cpp

```diff
@@ -0,0 +1,527 @@
+    const std::optional<std::string> inputFile = std::nullopt;
+    const std::optional<std::string> outputFile = std::nullopt;
+    const std::optional<std::string> phraseList;
+    const bool suppressOutputEnabled = false;
+    const bool partialResultsEnabled = false;
+    const std::optional<std::string> stablePartialResultThreshold = std::nullopt;
+    const bool srtEnabled = false;
+    const bool trueTextEnabled = false;
+    const std::string subscriptionKey;
+    const std::string region;
+    
+    UserConfig(
+        bool profanityFilterRemoveEnabled,
+        bool profanityFilterMaskEnabled,
+        std::optional<std::vector<std::string>> languageIDLanguages,
+        std::optional<std::string> inputFile,
+        std::optional<std::string> outputFile,
+        std::optional<std::string> phraseList,
+        bool suppressOutputEnabled,
+        bool partialResultsEnabled,
+        std::optional<std::string> stablePartialResultThreshold,
+        bool srtEnabled,
+        bool trueTextEnabled,
+        std::string subscriptionKey,
+        std::string region
+        ) :
+        profanityFilterRemoveEnabled(profanityFilterRemoveEnabled),
+        profanityFilterMaskEnabled(profanityFilterMaskEnabled),
+        languageIDLanguages(languageIDLanguages),
+        inputFile(inputFile),
+        outputFile(outputFile),
+        phraseList(phraseList),
+        suppressOutputEnabled(suppressOutputEnabled),
+        partialResultsEnabled(partialResultsEnabled),
+        stablePartialResultThreshold(stablePartialResultThreshold),
+        srtEnabled(srtEnabled),
+        trueTextEnabled(trueTextEnabled),
+        subscriptionKey(subscriptionKey),
+        region(region)
+        {}
+};
+
+struct Timestamp
+{
+    const int startHours;
+    const int endHours;
+    const int startMinutes;
+    const int endMinutes;
+    const float startSeconds;
+    const float endSeconds;
+
+    Timestamp(
+        int startHours,
+        int endHours,
+        int startMinutes,
+        int endMinutes,
+        float startSeconds,
+        float endSeconds
+    ) :
+    startHours(startHours),
+    endHours(endHours),
+    startMinutes(startMinutes),
+    endMinutes(endMinutes),
+    startSeconds(startSeconds),
+    endSeconds(endSeconds)
+    {}
+};
+
+class TooFewArguments : public std::exception
+{
+    public: const char* what() const throw()
+    {
+        return "Too few arguments.\n";
+    }
+};
+
+/*
+Helper functions
+*/
+
+/* Simple command line argument parsing functions. See
+https://stackoverflow.com/a/868894
+The value parameter to std::find must be std::string. Otherwise std::find will compare the pointer values rather than the string contents.
+*/
+std::optional<std::string> GetCmdOption(char** begin, char** end, const std::string& option)
+{
+    char** result = std::find(begin, end, option);
+    if (result != end)
+    {
+// We found the option (for example, "-o"), so advance from that to the value (for example, "filename").
```

Comment:
In our samples comments are aligned with the indentation. In Caption samples all comments starts from the beginning of line